### PR TITLE
add typescript definitions

### PR DESCRIPTION
**Pull request by bmagistro**
Reference: Add typescript definitions #341
Reference: Module 'moment' has no exported member 'MomentFormatSpecification' #343 

I couldn't come up with a solution to 343 other than copying lines 9-13 into the typescript file for moment-timezone.

Supports: https://github.com/grafana/clock-panel/pull/19

### CODE REVIEW

File: moment-timezone.d.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../moment/moment.d.ts" />
+
+declare namespace moment {
+  interface MomentBuiltinFormat {
+    __momentBuiltinFormatBrand: any;
+  }
+
+  type MomentFormatSpecification = string | MomentBuiltinFormat | (string | MomentBuiltinFormat)[];
```
Comment (westy92) [inline moment-timezone.d.ts]: Is the inconsistent indentation intentional?  (This and above vs. the interfaces below.)

Comment (westy92) [inline moment-timezone.d.ts]: Most of the file is indented correctly--just be consistent.

**Comment by westy92** [comment]
You should also add some tests like what moment has.

See https://github.com/moment/moment/tree/d28a819dc668f107566f306f827df07d43e30287/typing-tests and https://github.com/moment/moment/blob/d28a819dc668f107566f306f827df07d43e30287/Gruntfile.js#L173.

**Comment by bmagistro** [comment]
Not sure if this is something that can be added to the build deps to allow CI to continue, but there is a dependency on "grunt-exec" and "ntypescript".

Tests came directly from DefinitelyTyped.

**Comment by bmagistro** [comment]
Changes to ".travis.yml" align with the changes being requested in #310

**Comment by bmagistro** [comment]
Any updates on this getting merged in?

**Comment by maggiepint** [comment]
So, this situation has been nothing but heartbreak since we started it in the main moment library. Does this package currently work? https://www.npmjs.com/package/@types/moment-timezone

If so, strong preference for leaving them there.

**Comment by bmagistro** [comment]
At this point I don't remember if it worked and I couldn't figure it out or if it didn't work.  I am going to say I likely had an issue with the mentioned package likely similar if not the same as #343.

**Comment by bmagistro** [comment]
Closing PR, trying to redo the update and it appears this was resolved, likely in #343

**Pull request closed without merging**